src/revwalk.c

```diff
@@ -449,6 +453,45 @@ static int limit_list(git_commit_list **out, git_revwalk *walk, git_commit_list
 	return 0;
 }
 
+static int get_one_revision(git_commit_list_node **out, git_revwalk *walk, git_commit_list **list)
+{
+	int error;
+	git_commit_list_node *commit;
+
+	while(true) {
```

Comment:
This loop is useless right now -- we always return during the first loop

Reply:
Yeah, part of the porting. Here is where simplification handling happens in git. It's probably worth removing it to reduce confusion.